### PR TITLE
Feature - billing

### PR DESCRIPTION
**Pull request by joefitter**
* Added endpoint for listing establishment pils

needs:
https://github.com/UKHomeOffice/asl-schema/pull/241
and
https://github.com/UKHomeOffice/asl-permissions/pull/54

### CODE REVIEW

File: lib/routers/profile/pil.js [hunk 1]
```diff
@@ -88,6 +89,12 @@ const attachEstablishmentDetails = (req, res, next) => {
 };
 
 router.param('pilId', (req, res, next, id) => {
+  console.log({ id });
```
Comment (lennym) [inline lib/routers/profile/pil.js]:
```suggestion
```

File: lib/routers/profile/pil.js [hunk 2]
```diff
@@ -88,6 +89,12 @@ const attachEstablishmentDetails = (req, res, next) => {
...
+  if (id === 'transfers' || id === 'count') {
+    console.log('RETURNING');
```
Comment (lennym) [inline lib/routers/profile/pil.js]:
```suggestion
```

File: lib/routers/profile/pil.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const { get } = require('lodash');
+const activeBetween = require('@asl/schema/lib/active-between');
```
Comment (lennym) [inline lib/routers/profile/pil.js]: You've absolutely fucked me here.

File: lib/routers/profile/pil.js [hunk 4]
```diff
@@ -87,6 +87,58 @@ const attachEstablishmentDetails = (req, res, next) => {
     .catch(next);
 };
 
+const clean = (id, start, end) => pil => {
```
Comment (wheelsandcogs) [inline lib/routers/profile/pil.js]:
```suggestion
const removeSensitiveInfo = (id, start, end) => pil => {
```

File: lib/routers/profile/pil.js [hunk 5]
```diff
@@ -87,6 +87,58 @@ const attachEstablishmentDetails = (req, res, next) => {
...
+  // remove sensitive information from profile
```
Comment (wheelsandcogs) [inline lib/routers/profile/pil.js]:
```suggestion
```

File: lib/routers/profile/pil.js [hunk 6]
```diff
@@ -87,6 +87,58 @@ const attachEstablishmentDetails = (req, res, next) => {
...
+    let query = PIL.query().billable(params);
+    query = PIL.orderBy({ query, sort: { ...sort, column: sort.column || 'profile.lastName' } });
+    Promise.resolve()
+      .then(() => PIL.paginate({ query, limit, offset }))
+      .then(pils => {
+        res.meta.count = pils.total;
+        res.meta.total = pils.total;
+        res.meta.startDate = start;
+        res.meta.endDate = end;
+        res.response = pils.results.map(clean(req.establishment.id, start, end));
```
Comment (wheelsandcogs) [inline lib/routers/profile/pil.js]:
```suggestion
        res.response = pils.results.map(removeSensitiveInfo(req.establishment.id, start, end));
```

**Pull request merged**